src/main/java/org/logstash/plugins/inputs/http/util/SslSimpleBuilder.java

```diff
@@ -148,28 +204,80 @@ public boolean isClientAuthenticationRequired() {
     }
 
     public SslContext build() throws Exception {
+        if (this.trustStore != null && this.certificateAuthorities != null) {
+            throw new IllegalStateException("Use either an SSL certificate authorities or a Trust Store to configure client authentication");
+        }
+
         if (logger.isDebugEnabled()) {
             logger.debug("Available ciphers: {}", SUPPORTED_CIPHERS);
             logger.debug("Ciphers: {}", Arrays.toString(ciphers));
         }
 
-        SslContextBuilder builder = SslContextBuilder
-                .forServer(sslCertificateFile, sslKeyFile, passphrase)
+        SslContextBuilder builder = createSslContextBuilder()
                 .ciphers(Arrays.asList(ciphers))
-                .protocols(protocols);
+                .protocols(protocols)
+                .clientAuth(verifyMode.toClientAuth());
 
         if (isClientAuthenticationEnabled(verifyMode)) {
-            if (logger.isDebugEnabled()) {
-                logger.debug("Certificate Authorities: {}", Arrays.toString(certificateAuthorities));
-            }
+            if (certificateAuthorities != null) {
+                if (logger.isDebugEnabled()) {
+                    logger.debug("Certificate Authorities: {}", Arrays.toString(certificateAuthorities));
+                }
 
-            builder.clientAuth(verifyMode.toClientAuth())
-                    .trustManager(loadCertificateCollection(certificateAuthorities));
+                builder.trustManager(loadCertificateCollection(certificateAuthorities));
+            } else if (trustStore != null || keyStore != null) {
+                builder.trustManager(createTrustManagerFactory());
+            } else {
+                throw new IllegalStateException("Either an SSL certificate or an SSL Trust Store is required when SSL is enabled");
```

Comment:
I don't believe this should ever happen, right? if `ssl_enabled => true` then we already required the ssl_certificate or keystore to be set here:

```ruby
     unless ssl_certificate_configured? || ssl_keystore_configured?
      raise LogStash::ConfigurationError, "Either an `ssl_certificate` or `ssl_keystore_path` is required when SSL is enabled `#{ssl_config_name} => true`"
    end
```

Reply:
Exactly, it shouldn't happen, unless a bug is introduced. Same reasoning as [this](https://github.com/logstash-plugins/logstash-input-http/pull/171/files#r1406412378)